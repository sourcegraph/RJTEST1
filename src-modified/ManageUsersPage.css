.user {
    display: flex;
    align-items: center;
    justify-content: space-between;

    padding: 1rem;
    margin-bottom: 1rem;
    background-color: #e6ebf2;
}

/* User 'name' */
.user span {
    color: #0f111a;
}

/* User 'role' */
.user span ~ span {
    color: #5e6e8c;
}

.button {
    padding: 1rem;
    border-radius: 1rem;
    background-color: #c92a2a;
    color: #ffffff !important;
    cursor: pointer;
}
