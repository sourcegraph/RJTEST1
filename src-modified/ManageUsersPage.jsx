import React, { useEffect, useState } from 'react'

import { analytics, deleteUser, getUsers } from '../index.js'
import './ManageUsersPage.css'

const onDeleteUser = async (id) => {
    await deleteUser(id)

    // Remove user from list
    document.getElementById(`user-${id}`)?.remove()
}

const UserNode = ({ id, name, role }) => (
    <div className="user" id={`user-${id}`}>
        <div>
            <span>{name}</span> - <span>{role}</span>
        </div>
        <div className="button" onClick={() => onDeleteUser(id)}>
            Delete user
        </div>
    </div>
)

const UserList = ({ users }) => {
    if (users.length === 0) {
        return null
    }

    return (
        <>
            <div role="heading" style={{ fontSize: '1.5rem', fontWeight: 'bold', marginBottom: '1rem' }}>
                User List
            </div>
            {users.map(({ id, name, role }) => (
                <UserNode key={name} id={id} name={name} role={role} />
            ))}
        </>
    )
}

export const ManageUsersPage = () => {
    const [users, setUsers] = useState([])

    useEffect(() => {
        analytics.log('Manage users page visited')
    })

    useEffect(() => {
        getUsers().then(setUsers)
    }, [])

    return <UserList users={users} />
}
