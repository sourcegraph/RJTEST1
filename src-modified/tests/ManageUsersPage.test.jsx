import { render } from '@testing-library/react'
import { ManageUsersPage } from '../ManageUsersPage'

describe('ManageUsersPage', () => {
    it('renders correctly', async () => {
        const { findByText, container } = render(<ManageUsersPage />)
        const heading = await findByText('User List')
        expect(heading).toBeTruthy()
        expect(container).toMatchSnapshot()
    })
})
