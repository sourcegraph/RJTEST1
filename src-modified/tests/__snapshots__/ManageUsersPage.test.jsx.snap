// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageUsersPage renders correctly 1`] = `
<div>
  <div
    role="heading"
    style="font-size: 1.5rem; font-weight: bold; margin-bottom: 1rem;"
  >
    User List
  </div>
  <div
    class="user"
    id="user-abcd"
  >
    <div>
      <span>
        Alex
      </span>
       - 
      <span>
        Admin
      </span>
    </div>
    <div
      class="button"
    >
      Delete user
    </div>
  </div>
  <div
    class="user"
    id="user-efgh"
  >
    <div>
      <span>
        Erin
      </span>
       - 
      <span>
        Developer
      </span>
    </div>
    <div
      class="button"
    >
      Delete user
    </div>
  </div>
  <div
    class="user"
    id="user-ijkh"
  >
    <div>
      <span>
        Isaac
      </span>
       - 
      <span>
        Developer
      </span>
    </div>
    <div
      class="button"
    >
      Delete user
    </div>
  </div>
</div>
`;
